### PR TITLE
feat(functions): add generic types

### LINKED ISSUE #663: feat: cloud functions generic types
### Plugin(s)

- [ ] Analytics
- [ ] App
- [ ] App Check
- [ ] Authentication
- [ ] Crashlytics
- [ ] Cloud Firestore
- [X] Cloud Functions
- [ ] Cloud Messaging
- [ ] Cloud Storage
- [ ] Performance
- [ ] Remote Config

### Current problem

In the plugin, the request data and the response data are typed as any.

### Preferred solution

Like the JS SDK, we can add the generic type: `<RequestData = unknown, ResponseData = unknown>`

### Alternative options

_No response_

### Additional context

_No response_

### Before submitting

- [X] I have read and followed the [feature request guidelines](https://capawesome.io/contributing/feature-requests/).
- [X] I have attached links to possibly related issues and discussions.

### PR DESCRIPTION
**Pull request by mamillastre**
## Pull request checklist

Please check if your PR fulfills the following requirements:

- [x] The changes have been tested successfully.
- [x] A changeset has been created (`npm run changeset`).
- [x] I have read and followed the [pull request guidelines](https://capawesome.io/contributing/pull-requests/).

Close #663

### CODE REVIEW

File: packages/functions/src/definitions.ts [hunk 1]
```diff
@@ -54,28 +59,28 @@ export interface CallByUrlOptions extends CallOptions {
...
  */
-export interface CallOptions {
+export interface CallOptions<RequestData = unknown> {
   /**
    * The data to pass to the callable function.
    *
    * @since 6.1.0
    */
-  data?: any;
+  data?: RequestData | null;
```
Comment (robingenz) [inline packages/functions/src/definitions.ts:73]:
```suggestion
  data?: RequestData;
```

There is no reason for `null`, is there? If you don't need the param, you simply don't set it.

Comment (mamillastre) [reply packages/functions/src/definitions.ts:73]: I aligned the types to the js SDK:
```
export declare type HttpsCallable<RequestData = unknown, ResponseData = unknown> = (data?: RequestData | null) => Promise<HttpsCallableResult<ResponseData>>;
```
where the data can be null.

But you can tell me and I use `data?: RequestData;` instead.

Comment (robingenz) [reply packages/functions/src/definitions.ts:89]: Okay, in that case, we'll leave it like this for now.

**Review by robingenz — APPROVED** [review]
Thank you!

**Pull request merged**